hack/install-vm.sh

```diff
@@ -141,17 +150,246 @@ EOF
+            install_container_runtime
+        fi
+        # Install docker-compose
+        docker-compose version
+        if [ "$?" -ne 0 ]; then
+            cmd="curl -L "${DOCKER_COMPOSE}-$(uname -s)-$(uname -m)" -o ${DOCKER_COMPOSE_DIR}"
+            execute_command "$cmd" 1m
+            if [ "$?" -eq 0 ]; then
+                echo "docker-compose is downloaded successfully"
+                chmod +x ${DOCKER_COMPOSE_DIR}
+                ln -s ${DOCKER_COMPOSE_DIR} /usr/bin/docker-compose
+            else
+                echoerr "docker-compose couldn't be downloaded, exiting"
+                exit 2
+            fi
+        fi
+    fi
+    set -eo pipefail
+}
+
+# Create docker-compose file with ovs and antrea-agent container specific parameters
+create_docker_compose_file() {
+    cat << EOF > $AGENT_CONF_PATH/docker-compose.yaml
+version: "3"
+
+services:
+  ovs:
+    image: $ANTREA_DOCKER_IMAGE
+    container_name: ovs
+    volumes:
+        - /var/log/openvswitch:/var/log/openvswitch
+        - /var/run/openvswitch:/var/run/openvswitch
+        - /lib/modules:/lib/modules
+    command: $OVS_BIN
+    network_mode: host
+    cap_add:
+        - NET_ADMIN
+        - SYS_NICE
+        - SYS_ADMIN
+        - IPC_LOCK
+
+  antrea-agent:
+    image: $ANTREA_DOCKER_IMAGE
+    container_name: antrea-agent
+    volumes:
+        - /etc/antrea/:/etc/antrea
+        - /var/run/antrea:/var/run/antrea
+        - /var/run/openvswitch:/var/run/openvswitch
+        - /var/lib/cni:/var/lib/cni
+        - /var/log/antrea:/var/log/antrea
+        - /proc:/host/proc
+        - /var/run/netns:/host/var/run/netns
+        - /run/xtables.lock:/run/xtables.lock
+    environment:
+        - NODE_NAME=$NODE_NAME
+    command: $ANTREA_AGENT_BIN --config $AGENT_CONF_PATH/antrea-agent.conf --logtostderr=false --log_file=$AGENT_LOG_DIR/antrea-agent.log
+    network_mode: host
+    privileged: true
+    cap_add:
+        - SYS_MODULE
+EOF
+}
+
+load_ovs_modules() {
+    for module in "${OVS_MODULES[@]}"; do
+        lsmod | grep -q $module
+        if [[ $? -ne 0 ]]; then
+            echo "Load $module kernel module"
+            modprobe $module
+        fi
+    done
+}
+
+create_antrea_agent_watchdog_file() {
+    cat >$AGENT_CONF_PATH/antrea-agent-watchdog.sh << 'EOF'
+#!/bin/bash
+
+AGENT_CONF_PATH="/etc/antrea"
+AGENT_LOG_DIR="/var/log/antrea"
+
+declare -a OVS_KERNEL_MODULES=("vport_geneve" "openvswitch")
+declare -a DOCKER_SERVICE=("ovs" "antrea-agent")
+
+load_ovs_modules() {
+    echo "Loading OVS kernel"
+    for module in "${OVS_KERNEL_MODULES[@]}"; do
+        lsmod | grep -q $module
+        if [[ $? -ne 0 ]]; then
+            echo "Load $module kernel module"
+            modprobe $module
+        fi
+    done
+}
+
+exec > >(tee $AGENT_LOG_DIR/vm-watchdog.log) 2>&1
+
+docker info
+if [[ $? -ne 0 ]]; then
+    echo "Starting docker service"
+    systemctl start docker
+    sleep 2
+fi
+
+load_ovs_modules
+
+while(true); do
```

Comment:
watchdog should have a limit on the number of restart.. If agent is continuously crashing, we need to bail out eventually.. 500 to 1000 times restart is fine..